docs/cudf/source/guide-to-udfs.ipynb

```diff
@@ -0,0 +1,1708 @@
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>-0.691674</td>\n",
+       "      <td>True</td>\n",
+       "      <td>Dan</td>\n",
+       "      <td>-6.916743</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>0.480099</td>\n",
+       "      <td>False</td>\n",
+       "      <td>Bob</td>\n",
+       "      <td>4.800994</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>-0.473370</td>\n",
+       "      <td>True</td>\n",
+       "      <td>Xavier</td>\n",
+       "      <td>-4.733700</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>0.067479</td>\n",
+       "      <td>True</td>\n",
+       "      <td>Alice</td>\n",
+       "      <td>0.674788</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>-0.970850</td>\n",
+       "      <td>False</td>\n",
+       "      <td>Sarah</td>\n",
+       "      <td>-9.708501</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "          a      b       c         e\n",
+       "0 -0.691674   True     Dan -6.916743\n",
+       "1  0.480099  False     Bob  4.800994\n",
+       "2 -0.473370   True  Xavier -4.733700\n",
+       "3  0.067479   True   Alice  0.674788\n",
+       "4 -0.970850  False   Sarah -9.708501"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that, while we're operating on the Series `df['e']`, the kernel is actually executing on the [DeviceNDArray](https://numba.pydata.org/numba-doc/dev/cuda/memory.html#device-arrays) \"underneath\" the Series. If you ever need to grab the underlying DeviceNDArray of a Series, you can do so like this: `Series.data.mem`. We'll see an example of doing this in the Null Handling section of this guide."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# DataFrame UDFs\n",
+    "\n",
+    "We could apply a UDF on a DataFrame just like we did above with `forall`. We just need to write a kernel that expects multiple inputs, and pass multiple Series as arguments when we execute our kernel. Because this is fairly common and can get a bit difficult to manage, cuDF provides two APIs to streamline this: `apply_rows` and `apply_chunks`. Below, we walk through an example of using `apply_rows` (`apply_chunks` works in a similar way--it just also gives you more control over low-level kernel behavior. See the [documentation](https://rapidsai.github.io/projects/cudf/en/0.10.0/api.html#cudf.dataframe.DataFrame.apply_chunks) for information).\n",
```

Comment:
```suggestion
    "We could apply a UDF on a DataFrame like we did above with `forall`. We'd need to write a kernel that expects multiple inputs, and pass multiple Series as arguments when we execute our kernel. Because this is fairly common and can be difficult to manage, cuDF provides two APIs to streamline this: `apply_rows` and `apply_chunks`. Below, we walk through an example of using `apply_rows` (`apply_chunks` works in a similar way--it just also offers more control over low-level kernel behavior. See the [documentation](https://rapidsai.github.io/projects/cudf/en/0.10.0/api.html#cudf.dataframe.DataFrame.apply_chunks) for information).\n",
```